mailbox/api/src/main/java/org/apache/james/mailbox/model/SimpleMailboxAnnotation.java

```diff
@@ -0,0 +1,328 @@
+package org.apache.james.mailbox.model;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Pattern;
+
+import org.apache.james.mailbox.exception.UnsupportedAnnotationException;
+import org.apache.james.mailbox.model.MailboxACL.MailboxACLEntryKey;
+
+import com.google.common.base.Preconditions;
+
+public class SimpleMailboxAnnotation implements MailboxAnnotation {
+
+    private static final Pattern NAME_ANNOTATION_PATTERN = Pattern.compile("[^\\x1a-\\x7f]");
```

Comment:
I did not get that Pattern from internet. I just read the RFC said "NOT contain non-ASCII characters or characters with octet values in the range 0x00 to 0x19"
The ASCII is 0x00 to 0x7f so ("[^\\x1a-\\x7f]") = (0x10-0x7f) - (0x10-0x19)
Is that ok?